src/mon/OSDMonitor.cc

```diff
@@ -3144,7 +3146,11 @@ void OSDMonitor::do_application_enable(int64_t pool_id,
     p = pending_inc.new_pools[pool_id];
   }
 
-  p.application_metadata.insert({app_name, {}});
+  if (app_key.empty()) {
+    p.application_metadata.insert({app_name, {}});
+  } else {
+    p.application_metadata.insert({app_name, {{app_key, app_value}}});
```

Comment:
Again, doesn’t this add empty entries in the default case? I think we want to only add the key/value conditionally if they have length?

Reply:
No, I'm pretty sure that's correct. It's exactly the behavior from before when they key is empty.

Reply:
Previously we were passing in an empty map. Now we are passing in a single-element map whose entry has a key and a value both consisting of an empty string.

Reply:
@fullerdj ^

Reply:
Only if the key is nonempty. I changed it based on your earlier comment.

Reply:
Oh dur, thanks.